contracts/facets/EmbalmerFacet.sol

```diff
@@ -34,267 +34,237 @@ contract EmbalmerFacet {
         if (selectedArchaeologists.length == 0) {
             revert LibErrors.NoArchaeologistsProvided();
         }
-
-        // Confirm that minShards is greater than 0
-        if (sarcophagus.minShards == 0) {
+        // Confirm that threshold is greater than 0
+        if (sarcophagusParams.threshold == 0) {
             revert LibErrors.MinShardsZero();
         }
-
-        // Confirm that minShards is less than or equal to the number of archaeologists
+        // Confirm that threshold is less than or equal to the number of archaeologists
         // (k <= n in a shamir secret sharing scheme)
-        if (sarcophagus.minShards > selectedArchaeologists.length) {
-            revert LibErrors.MinShardsGreaterThanArchaeologists(
-                sarcophagus.minShards
-            );
+        if (sarcophagusParams.threshold > selectedArchaeologists.length) {
+            revert LibErrors.MinShardsGreaterThanArchaeologists(sarcophagusParams.threshold);
         }
 
-        // Initialize a list of archaeologist addresses to be passed in to the
-        // sarcophagus object
-        address[] memory cursedArchaeologists = new address[](
-            selectedArchaeologists.length
-        );
-
+        // create the sarcophagus
+        LibTypes.Sarcophagus storage sarcophagus = s.sarcophagi[sarcoId];
+        sarcophagus.name = sarcophagusParams.name;
+        sarcophagus.threshold = sarcophagusParams.threshold;
+        sarcophagus.resurrectionTime = sarcophagusParams.resurrectionTime;
+        sarcophagus.maximumRewrapInterval = sarcophagusParams.maximumRewrapInterval;
+        sarcophagus.arweaveTxIds = arweaveTxIds;
+        sarcophagus.embalmerAddress = msg.sender;
+        sarcophagus.recipientAddress = sarcophagusParams.recipientAddress;
+        sarcophagus.cursedArchaeologistAddresses = new address[](selectedArchaeologists.length);
+
+        // track total digging fees due upon creation of sarcophagus
         uint256 totalDiggingFees = 0;
 
         for (uint256 i = 0; i < selectedArchaeologists.length; i++) {
-            LibTypes.SelectedArchaeologistData memory arch = selectedArchaeologists[i];
-            LibUtils.revertIfArchProfileDoesNotExist(arch.archAddress);
-
-            // Confirm that the archaeologist list is unique. This is done by
-            // checking that the archaeologist does not already exist from
-            // previous iterations in this loop.
-            if (LibUtils.archaeologistExistsOnSarc(sarcoId, arch.archAddress)) {
-                revert LibErrors.ArchaeologistListNotUnique(
-                    cursedArchaeologists
-                );
+            // confirm archaeologist is registered
+            LibUtils.revertIfArchProfileDoesNotExist(selectedArchaeologists[i].archAddress);
+
+            // Confirm archaeologist isn't already cursed on sarcophagus
+            // todo: may be unnecessary, is cursing an archaeologist twice harming anybody but the caller?
```

Comment:
Would it not hurt the archaeologist?

Reply:
It would just result in them getting their bond locked twice and their digging fees paid twice - so equivalent to the archaeologist getting cursed on two sarcophagi with the same digging fees

Reply:
Ah, true.

Although... it wouldn't be what the archaeologist explicitly signed up for. If we give them an interface for checking out their current responsibilities, they'll see 1 sarco, but total locked bond for 2. Probably won't have any way of knowing they'll be paid twice during rewrap/unwrap in the future.

Even if after the initial confusion it becomes a pleasant surprise, it _is_ a "bug" and might be abused in ways we've not even thought of

Reply:
```
sarcophagus.cursedArchaeologists[selectedArchaeologists[i].archAddress] = LibTypes
                .CursedArchaeologist({
                    isAccused: false,
                    diggingFee: selectedArchaeologists[i].diggingFee,
                    doubleHashedKeyShare: selectedArchaeologists[i].doubleHashedKeyShare,
                    rawKeyShare: ""
                });
```

It seems like if we didn't check this, then the `cursedArchaeologists` could get out of sync with `cursedArchaeologistAddresses`. Bc you could be cursing with 2 different keyshares on the same arch, but `cursedArchaeologistAddresses` would only save one of them (since its indexed by archAddress)

Reply:
Oh you're right, this would cause problems with the archaeologist not being able to unlock their bonds twice - will remove this todo. So far I've kept all validation in the contracts except the check for arweave tx ids which is getting enforced through typing now. I'm actually going to remove some of these validation todos because I think the possibility of some nonstandard state causing unexpected behavior isn't worth the gas savings with most of these. When we get to the gas cost evaluation task we may revisit this idea.